azurerm/internal/services/web/app_service_slot_custom_hostname_binding_resource.go

```diff
@@ -0,0 +1,213 @@
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_app_service_slot_custom_hostname_binding", *existing.ID)
+		}
+	}
+
+	properties := web.HostNameBinding{
+		HostNameBindingProperties: &web.HostNameBindingProperties{
+			SiteName: utils.String(appServiceName),
+		},
+	}
+
+	if sslState != "" {
+		if thumbprint == "" {
+			return fmt.Errorf("`thumbprint` must be specified when `ssl_state` is set")
+		}
+
+		properties.HostNameBindingProperties.SslState = web.SslState(sslState)
+	}
+
+	if thumbprint != "" {
+		if sslState == "" {
+			return fmt.Errorf("`ssl_state` must be specified when `thumbprint` is set")
+		}
+
+		properties.HostNameBindingProperties.Thumbprint = utils.String(thumbprint)
+	}
+
+	if _, err := client.CreateOrUpdateHostNameBindingSlot(ctx, resourceGroup, appServiceName, hostname, properties, appServiceSlot); err != nil {
+		return fmt.Errorf("Error creating/updating Custom Hostname Binding %q (App Service %q / Slot %q / Resource Group %q): %+v", hostname, appServiceName, appServiceSlot, resourceGroup, err)
+	}
+
+	read, err := client.GetHostNameBindingSlot(ctx, resourceGroup, appServiceName, appServiceSlot, hostname)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Custom Hostname Binding %q (App Service %q / Slot %q / Resource Group %q): %+v", hostname, appServiceName, appServiceSlot, resourceGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Hostname Binding %q (App Service %q / Slot %q / Resource Group %q) ID", hostname, appServiceName, appServiceSlot, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceAppServiceSlotCustomHostnameBindingRead(d, meta)
+}
+
+func resourceAppServiceSlotCustomHostnameBindingRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AppServiceSlotCustomHostnameBindingID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetHostNameBindingSlot(ctx, id.ResourceGroup, id.AppServiceName, id.AppServiceSlot, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] App Service Hostname Binding %q (App Service %q / Slot %q / Resource Group %q) was not found - removing from state", id.Name, id.AppServiceName, id.AppServiceSlot, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving Custom Hostname Binding %q (App Service %q / Slot %q / Resource Group %q): %+v", id.Name, id.AppServiceName, id.AppServiceSlot, id.ResourceGroup, err)
+	}
+
+	d.Set("hostname", id.Name)
+	d.Set("app_service_name", id.AppServiceName)
+	d.Set("app_service_slot", id.AppServiceSlot)
+	d.Set("resource_group_name", id.ResourceGroup)
+
+	if props := resp.HostNameBindingProperties; props != nil {
+		d.Set("ssl_state", props.SslState)
```

Comment:
This may need casting to a string?
```suggestion
		d.Set("ssl_state", string(props.SslState))
```